sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala

```diff
@@ -403,7 +403,41 @@ class predicateJoinTestScala extends TestBaseScala {
         assert(distanceJoinDf.queryExecution.sparkPlan.collect { case p: DistanceJoinExec => p }.size === 1)
         assert(distanceJoinDf.count() == expected)
       })
+    }
+
+    it("Passed ST_HausdorffDistance in a spatial join") {
+      val sampleCount = 100
+      val distanceCandidates = Seq(1, 2)
+      val densityFrac = 0.6
+      val inputPoint = buildPointDf.limit(sampleCount).repartition(5)
+      val inputPolygon = buildPolygonDf.limit(sampleCount).repartition(3)
 
+      distanceCandidates.foreach(distance => {
+
+        //DensityFrac specified, <= distance
+        val expectedDensityIntersects = bruteForceDistanceJoinHausdorff(sampleCount, distance, densityFrac, true)
+        val distanceDensityIntersectsDF = inputPoint.alias("pointDF").join(inputPolygon.alias("polygonDF"), expr(s"ST_HausdorffDistance(pointDF.pointshape, polygonDF.polygonshape, $densityFrac) <= $distance"))
+        assert(distanceDensityIntersectsDF.queryExecution.sparkPlan.collect { case p: DistanceJoinExec => p }.size === 1)
+        assert(distanceDensityIntersectsDF.count() == expectedDensityIntersects)
```

Comment:
I wonder what the number of expected intersects is in these test cases? Can you report the numbers here?

Reply:
@iGN5117

Reply:
Can you add more distanceCandidates? Most importantly, add some candidates that lead to results > 100.

Reply:
Sure

Reply:
Have added candidates 1, 2, 5, 10. These yield result 100, 298, 688, 1258 respectively for a sample size of 100